### PR TITLE
internal/v4: add debug/pprof endpoint

### PR DESCRIPTION
**Pull request by rogpeppe**
This will make it possible to gain insight into how the server is behaving at runtime.

I've added a test for the endpoint that's most straightforward to test.
The others will need to be QA'd.

To QA, run the charmstore, then:

```
go tool pprof http://$charmstorehost/v4/debug/pprof/heap
```

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/charmstore/625/

File: internal/v4/api.go [hunk 1]
```diff
@@ -41,14 +42,18 @@ func New(store *charmstore.Store, config charmstore.ServerParams) *Handler {
...
-			"debug":              http.HandlerFunc(h.serveDebug),
-			"debug/status":       router.HandleJSON(h.serveDebugStatus),
-			"log":                router.HandleErrors(h.serveLog),
-			"search":             router.HandleJSON(h.serveSearch),
-			"search/interesting": http.HandlerFunc(h.serveSearchInteresting),
-			"stats/":             router.NotFoundHandler(),
-			"stats/counter/":     router.HandleJSON(h.serveStatsCounter),
+			"changes/published":   router.HandleJSON(h.serveChangesPublished),
+			"debug":               http.HandlerFunc(h.serveDebug),
+			"debug/pprof/":        http.HandlerFunc(pprof.Index),
```
Comment (mhilton) [inline internal/v4/api.go]: Could we put these behind the authentication? I'm not sure we want just anybody grabbing profile info.

**Comment by mhilton** [comment]
This doesn't appear to be working properly. If I go to http://localhost:8080/v4/debug/pprof/ and then click on any link I get a 404. It seems to have all of it's links as "/debug/pprof/" rather than "/v4/debug/pprof"

**Comment by rogpeppe** [comment]
Fixed the above issues.

Unfortunately there is one issue that I have not been able to fix: if you visit v4/debug/pprof,
it returns a redirect to debug/pprof/, which is wrong. This is because of the fact
that we strip prefixes in the handlers, so there is no way for the redirect handler
to generate the absolute URL needed for the Location header.
I'll keep mulling on this, but for the moment I don't think it should need to block
this branch.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/charmstore/645/

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/charmstore/646/

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/charmstore/648/

**Comment by urosj** [comment]
:+1: Did QA, worked fine.

**Comment by mhilton** [comment]
:+1:

**Comment by rogpeppe** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Pull request merged**